### ISSUE TITLE
Add a Sponsorship model

### ISSUE THREAD

**Issue by narayanaditya95** [swcarpentry/amy#830]
Introduce a M2M field between an `Organization` (a private org probably) and the `Event` model through a `Sponsorship` model.

The `Sponsorship` model would keep record of the kind and amount of sponsorship provided by an org.

**Comment by narayanaditya95** [comment]
@gvwilson The `Membership` model already models sponsorship from a host. Apart from an IntegerField to account for the exact amount of sponsorship provided (or a slab), it addresses most of things required for a `Sponsorship` model.

**Comment by narayanaditya95** [comment]
@gvwilson ping.

**Comment by gvwilson** [comment]
I vote for augmenting the existing model (in the same way that we augmented Task with a title field). @pbanaszkiewicz?

**Comment by narayanaditya95** [comment]
The `Membership` model can easily be extended to show PyData sponsors.
A PyData sponsor looks like:
```json
{
    "name": "Continuum Analytics",
    "external_url": "http://continuum.io/",
    "level": "PyData Founding",
    "annotation": "Continuum Analytics develops Anaconda, the leading modern open source analytics platform powered by Python. Continuum’s Python-based platform and consulting services empower organizations to analyze, manage and visualize big data - turning massive datasets into actionable insights and business value. Built on proven open-source technology and easily integrated within existing IT environments, Anaconda allows organizations to make critical business decisions based on their data quickly, easily, inexpensively, and with flexibility. Without having to worry about how to access their data, organizations can free up resources to solve actual problems. Continuum’s founders and developers have created or contribute to some of the most popular data science technologies, including NumPy, SciPy, Pandas, IPython, and many others. To learn more about the Anaconda platform, training and consulting services, visit continuum.io."
}
```

The corresponding `Membership` object would be:
```py
Membership(
    variant='Sponsor',
    agreement_start=null,
    agreement_end=null,
    contribution_type="PyData Founding",
    workshops_without_admin_fee_per_year=null,
    self_organized_workshops_per_year=null,
    notes ='Continuum Analytics develops Anaconda....',
    organization = Organization('Continuum', 'continuum.io') #after migration to Org
)
```

The required  changes are:
- `Membership.CONTRIBUTION_CHOICES` must be extended to include
```
PyData Founding
PyData Diamond
PyData Platinum
PyData Gold
PyData Silver
PyData Supporting
PyData Community
```

**Comment by narayanaditya95** [comment]
@gvwilson 
After having a talk with @pbanaszkiewicz yesterday, we realized that the `Membership` model in AMY serves a different use case than what the `Sponsorship` model in PyData does.

The `Membership` model in AMY records the sponsorship of an organization to the SWC/DC organization as a whole.
The `Sponsorship` model in PyData records the sponsorship provided by an organization to each particular conference, and not to PyData as a whole. See [this](http://2015.pydata.fr/static/pdf/pydata-paris-2016-sponsorship.pdf)

There is no field in the `Membership` model that relates to an `Event` object and they also serve different purposes. Thus, @pbanaszkiewicz and I decided to create a new model in itself.
